goalie/go_mesh_seq.py

```diff
@@ -228,6 +232,17 @@ def indicate_errors(
                     indi.interpolate(abs(indi))
                     self.indicators[f][i][j].interpolate(ufl.max_value(indi, 1.0e-16))
 
+            # discard extra subinterval solution field
+            self.solutions[f][FWD_OLD].pop()
+            self.solutions[f][ADJ_NEXT].pop()
+            enriched_mesh_seq.solutions[f][FWD_OLD].pop()
+            enriched_mesh_seq.solutions[f][ADJ_NEXT].pop()
+
+            # delete extra mesh to reduce the memory footprint
+            enriched_mesh_seq.meshes.pop()
```

Comment:
I think you also need to remove function spaces in `enriched_mesh_seq._fs` in order to fully get rid of the mesh. Maybe somewhere else too

Reply:
see commit 714f9ef - added call to `update_function_spaces` which uses the existing `MeshSeq` method to rebuild `enriched_mesh_seq._fs` based on the updated subintervals.